mm2src/coins/lp_coins.rs

```diff
@@ -1913,15 +1913,33 @@ pub trait MarketCoinOps {
     fn is_trezor(&self) -> bool;
 }
 
+/// Priority levels for UTXO fee estimation for withdrawal.
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub enum UtxoFeePriority {
+    /// Low priority: Estimated confirmation within approximately 3 blocks.
+    Low,
+    /// Normal priority: Estimated confirmation within approximately 2 blocks.
+    Normal,
+    /// High priority: Estimated confirmation within approximately 1 block.
+    High,
+}
+
 #[derive(Clone, Debug, Deserialize, PartialEq)]
 #[serde(tag = "type")]
 pub enum WithdrawFee {
+    /// encapsulates the fixed fee amount for a withdrawal transaction, regardless of transaction size.
     UtxoFixed {
         amount: BigDecimal,
     },
+    /// encapsulates the fee amount for a withdrawal transaction calculated based on the transaction size in kilobytes.
     UtxoPerKbyte {
         amount: BigDecimal,
     },
+    /// encapsulates the priority of a withdrawal transaction, indicating the desired fee
+    /// level for transaction processing.
+    UtxoPriority {
+        priority: UtxoFeePriority,
+    },
```

Comment:
Here is how I envision the GUI implementation of this part. 

We present the user with the three levels of fees and let them make a choice. The GUI would potentially make three separate withdrawal requests, each using a different priority level. This would generate three different raw transaction hexes, each associated with a different fee and expected confirmation time. 

Once the user selects the fee they prefer, the corresponding transaction hex would be used. 

What are your thoughts on this approach? Would it be feasible to make some changes that simplifies the process for the GUI to handle this with a single request?

Reply:
Okay, here's mine view.

The api receives a priority withdrawal request from the gui, the api generates fee for the available priorities(currently 3) and send the response (each fee with it's tx hex) and finally, gui can render this three options for user to pick a choice.

Reply:
> The api receives a priority withdrawal request from the gui, the api generates fee for the available priorities(currently 3) and send the response (each fee with it's tx hex) and finally, gui can render this three options for user to pick a choice.

This is how I see it as well, but it might be too complicated as we would probably need to return 3 `TransactionDetails` objects in the response which is different from the current withdraw response.

Reply:
FYI for eth I also added priority levels for tx fees:
For withdrawals user can add a new WithdrawFee option and set explicit fee per gas. 
For swaps there is an rpc to set the desired level of low/medium/priority for the whole swap (so the API will by itself query the current fee value before each swap step).
There is also an rpc for GUI to query fees for all supported 3 levels (to show fe values to the user so he could select desired level). Initially user should start a loop in the API which periodically calculates priority fees, so the query rpc just immediately returns the recent estimated fee values.

Maybe we could do this similar both for utxo and eth.

Reply:
(Some explanation why a there is loop in API to obtain fees. This is how other eth wallet apps do it: API does fee estimations getting data from the eth provider. The GUI subscribes to the fee updates from API so it can in realtime notify the user about changes)